external-builds/pytorch/run_linux_pytorch_tests.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -uxo pipefail
+
+# Validate environment variable
+if [ -z "$PYTHON_VERSION" ]; then
+  exit 1
+fi
+
+# Determine directories
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="$(realpath "$SCRIPT_DIR/../..")"
+PYTORCH_DIR="$ROOT_DIR/external-builds/pytorch/pytorch"
+
+# Activate virtual environment
+VENV_DIR="$ROOT_DIR/venv"
+if [ ! -d "$VENV_DIR" ]; then
+  exit 1
+fi
+source "$VENV_DIR/bin/activate" || exit 1
+
+# Set trap to deactivate virtual environment
+trap "deactivate 2> /dev/null" EXIT
+
+# Checkout repositories
+cd "$ROOT_DIR"
+./external-builds/pytorch/pytorch_torch_repo.py checkout --no-hipify --no-patch
```

Comment:
This could also be the responsibility of the workflow rather than the test script. If I tried to run this as a developer, this would overwrite my pytorch checkout.